frontend/src/api/users.ts

```diff
@@ -209,4 +210,33 @@ export const UsersService = {
       staleTime: 60e3,
     });
   },
+
+  useReportUser: <T = void>(
+    mutationOptions?: Omit<
+      UseMutationOptions<T, Error, ReportUserData>,
+      "mutationFn" | "onSuccess"
+    >
+  ) => {
+    const axios = useAxiosPrivate();
+    const queryClient = useQueryClient();
+    return useMutation<T, Error, ReportUserData>({
+      mutationFn: async (data) => {
+        const response = await axios.post<T>(
+          "/reports/add",
+          {},
+          { params: data }
+        );
+        return response.data;
+      },
+      onSuccess: () => {
+        return queryClient.invalidateQueries({
+          queryKey: [
+            "UsersService.getCurrentUser",
+            "MatchesService.getMatches",
+          ],
+        });
+      },
```

Comment:
https://stackoverflow.com/questions/74370694/react-query-invalidate-multiple-queries-but-wait-until-they-all-finished-on-mut
```suggestion
      onSuccess: () => Promise.all([
        queryClient.invalidateQueries({
          queryKey: ["UsersService.getCurrentUser"],
        }),
        queryClient.invalidateQueries({
          queryKey: ["MatchesService.getPotentialUsers"],
        }),
      ]),
```